text/0000-rustflags.md

```diff
@@ -0,0 +1,289 @@
+
+## --rustflag
+
+The `--rustflag` option will allow a Rust developer to pass any rustc flag to the root crate being built. This will allow a simple
+command such as `cargo test` to have the option of setting the `-Cinstrument-coverage` flag for a single crate and run the unit tests
+ensuring coverage data is collected for this crate and this crate only. For example, let's take crate `foo`:
+
+Cargo.toml:
+```toml
+[package]
+name = "foo"
+version = "0.1.0"
+```
+
+cargo CLI (Output lines have been removed for simplicity):
+```
+cargo test --rustflag=-Cinstrument-coverage
+
+Compiling foo v0.1.0 (...)
+     Running `rustc --crate-name foo --edition=2021 src\lib.rs --crate-type lib ... -Cinstrument-coverage`
+     Running `rustc --crate-name foo --edition=2021 src\lib.rs --test ... -Cinstrument-coverage`
+    Finished test [unoptimized + debuginfo] target(s) in 1.35s
+     Running `target/debug/deps/foo-669448d9b4043564`
+
+running 1 test
+test tests::it_works ... ok
+
+test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
+```
+
+Running this command will build the crate `foo` with the flag `-Cinstrument-coverage` passed to the invocation of rustc
+for the crate `foo`. Dependencies such as the standard libraries and other upstream dependencies would not be instrumented
+saving on compilation time.
+
+## --rustflag for a workspace
+
+When building a workspace with multiple members, any `--rustflag=<RUSTFLAG>` options set will be passed to the invocation
+of the Rust compiler for all members unless `workspace.default-members` manifest key has been set. In that case, only the default
+members being compiled will have the rustflag options specified passed through to rustc. For example:
+
+Cargo.toml:
+```toml
+[workspace]
+members = ["foo", "bar"]
+```
+
+cargo CLI (Output lines have been removed for simplicity):
+```
+cargo build --rustflag=-Cinstrument-coverage
+
+Compiling foo v0.1.0 (.../foo/foo)
+Compiling bar v0.1.0 (.../foo/bar)
+    Running `rustc --crate-name foo ... -Cinstrument-coverage`
+    Running `rustc --crate-name bar ... -Cinstrument-coverage`
+Finished dev [unoptimized + debuginfo] target(s) in 0.98s
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+As mentioned above a new Cargo option, `--rustflag=<RUSTFLAG>`, would be added to several of the existing cargo subcommands.
+Those subcommands would be, `bench, build, check, run` and `test`. The `--rustflag=<RUSTFLAG>` option will require an `=`
+sign between the option name and the option value, due to the way Cargo parses the options passed to its subcommands. Cargo under
+the covers uses the `clap` crate to parse the command line invocation and set the relevant options passed to it. Since all rust
+compiler flags start with a `-`, without the `=` delimitter `clap` parses the rustc flag as a new Cargo flag instead leading to
+an error from Cargo.
```

Comment:
I think the best way to move forward is to remove the requirement of having the `=` delimitter. Since this uses clap as the underlying argument parser for cargo commands, clap has the option to allow a hyphen `-` to be used as a value to an option which is what is needed here.

This means that `clap` will also need a way to terminate multiple values for a given command but this is also supported via the `value_terminator` option. In this case it will be a `;`. This will make setting rustflags simplier by supporting:

```
cargo test --rustflags -C instrument-coverage -C strip=debuginfo -C save-temps ; -v -Z unstable-options
```

`-C instrument-coverage -C strip=debuginfo -C save-temps` would be flags passed to rustc and `-v -Z unstable-options` would be cargo flags.

I'll update the RFC to describe this.